File: generated-sources/java-vertx/mojang-authentication/src/main/java/org/openapitools/server/api/model/GameProfile.java
package org.openapitools.server.api.model;

import java.util.Objects;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonProperty;
import java.io.File;

@JsonInclude(JsonInclude.Include.NON_NULL) 
public class GameProfile   {
  
  private String agent = ;
  private String id = ;
  private String name = ;
  private File userId = ;
  private Long createdAt = ;
  private Boolean legacyProfile = ;
  private Boolean suspended = ;
  private Boolean paid = ;
  private Boolean migrated = ;
  private Boolean legacy = ;

  public GameProfile () {

  }

  public GameProfile (String agent, String id, String name, File userId, Long createdAt, Boolean legacyProfile, Boolean suspended, Boolean paid, Boolean migrated, Boolean legacy) {
    this.agent = agent;
    this.id = id;
    this.name = name;
    this.userId = userId;
    this.createdAt = createdAt;
    this.legacyProfile = legacyProfile;
    this.suspended = suspended;
    this.paid = paid;
    this.migrated = migrated;
    this.legacy = legacy;
  }

    
  @JsonProperty("agent")
  public String getAgent() {
    return agent;
  }
  public void setAgent(String agent) {
    this.agent = agent;
  }

    
  @JsonProperty("id")
  public String getId() {
    return id;
  }
  public void setId(String id) {
    this.id = id;
  }

    
  @JsonProperty("name")
  public String getName() {
    return name;
  }
  public void setName(String name) {
    this.name = name;
  }

    
  @JsonProperty("userId")
  public File getUserId() {
    return userId;
  }
  public void setUserId(File userId) {
    this.userId = userId;
  }

    
  @JsonProperty("createdAt")
  public Long getCreatedAt() {
    return createdAt;
  }
  public void setCreatedAt(Long createdAt) {
    this.createdAt = createdAt;
  }

    
  @JsonProperty("legacyProfile")
  public Boolean getLegacyProfile() {
    return legacyProfile;
  }
  public void setLegacyProfile(Boolean legacyProfile) {
    this.legacyProfile = legacyProfile;
  }

    
  @JsonProperty("suspended")
  public Boolean getSuspended() {
    return suspended;
  }
  public void setSuspended(Boolean suspended) {
    this.suspended = suspended;
  }

    
  @JsonProperty("paid")
  public Boolean getPaid() {
    return paid;
  }
  public void setPaid(Boolean paid) {
    this.paid = paid;
  }

    
  @JsonProperty("migrated")
  public Boolean getMigrated() {
    return migrated;
  }
  public void setMigrated(Boolean migrated) {
    this.migrated = migrated;
  }

    
  @JsonProperty("legacy")
  public Boolean getLegacy() {
    return legacy;
  }
  public void setLegacy(Boolean legacy) {
    this.legacy = legacy;
  }


  @Override
  public boolean equals(Object o) {
    if (this == o) {
      return true;
    }
    if (o == null || getClass() != o.getClass()) {
      return false;
    }
    GameProfile gameProfile = (GameProfile) o;
    return Objects.equals(agent, gameProfile.agent) &&
        Objects.equals(id, gameProfile.id) &&
        Objects.equals(name, gameProfile.name) &&
        Objects.equals(userId, gameProfile.userId) &&
        Objects.equals(createdAt, gameProfile.createdAt) &&
        Objects.equals(legacyProfile, gameProfile.legacyProfile) &&
        Objects.equals(suspended, gameProfile.suspended) &&
        Objects.equals(paid, gameProfile.paid) &&
        Objects.equals(migrated, gameProfile.migrated) &&
        Objects.equals(legacy, gameProfile.legacy);
  }

  @Override
  public int hashCode() {
    return Objects.hash(agent, id, name, userId, createdAt, legacyProfile, suspended, paid, migrated, legacy);
  }

  @Override
  public String toString() {
    StringBuilder sb = new StringBuilder();
    sb.append("class GameProfile {\n");
    
    sb.append("    agent: ").append(toIndentedString(agent)).append("\n");
    sb.append("    id: ").append(toIndentedString(id)).append("\n");
    sb.append("    name: ").append(toIndentedString(name)).append("\n");
    sb.append("    userId: ").append(toIndentedString(userId)).append("\n");
    sb.append("    createdAt: ").append(toIndentedString(createdAt)).append("\n");
    sb.append("    legacyProfile: ").append(toIndentedString(legacyProfile)).append("\n");
    sb.append("    suspended: ").append(toIndentedString(suspended)).append("\n");
    sb.append("    paid: ").append(toIndentedString(paid)).append("\n");
    sb.append("    migrated: ").append(toIndentedString(migrated)).append("\n");
    sb.append("    legacy: ").append(toIndentedString(legacy)).append("\n");
    sb.append("}");
    return sb.toString();
  }

  /**
   * Convert the given object to string with each line indented by 4 spaces
   * (except the first line).
   */
  private String toIndentedString(Object o) {
    if (o == null) {
      return "null";
    }
    return o.toString().replace("\n", "\n    ");
  }
}

File: generated-sources/cpp-tizen/mojang-authentication/src/GameProfile.cpp
#include <map>
#include <cstdlib>
#include <glib-object.h>
#include <json-glib/json-glib.h>
#include "Helpers.h"


#include "GameProfile.h"

using namespace std;
using namespace Tizen::ArtikCloud;

GameProfile::GameProfile()
{
	//__init();
}

GameProfile::~GameProfile()
{
	//__cleanup();
}

void
GameProfile::__init()
{
	//agent = std::string();
	//id = std::string();
	//name = std::string();
	//userId = std::string();
	//createdAt = long(0);
	//legacyProfile = bool(false);
	//suspended = bool(false);
	//paid = bool(false);
	//migrated = bool(false);
	//legacy = bool(false);
}

void
GameProfile::__cleanup()
{
	//if(agent != NULL) {
	//
	//delete agent;
	//agent = NULL;
	//}
	//if(id != NULL) {
	//
	//delete id;
	//id = NULL;
	//}
	//if(name != NULL) {
	//
	//delete name;
	//name = NULL;
	//}
	//if(userId != NULL) {
	//
	//delete userId;
	//userId = NULL;
	//}
	//if(createdAt != NULL) {
	//
	//delete createdAt;
	//createdAt = NULL;
	//}
	//if(legacyProfile != NULL) {
	//
	//delete legacyProfile;
	//legacyProfile = NULL;
	//}
	//if(suspended != NULL) {
	//
	//delete suspended;
	//suspended = NULL;
	//}
	//if(paid != NULL) {
	//
	//delete paid;
	//paid = NULL;
	//}
	//if(migrated != NULL) {
	//
	//delete migrated;
	//migrated = NULL;
	//}
	//if(legacy != NULL) {
	//
	//delete legacy;
	//legacy = NULL;
	//}
	//
}

void
GameProfile::fromJson(char* jsonStr)
{
	JsonObject *pJsonObject = json_node_get_object(json_from_string(jsonStr,NULL));
	JsonNode *node;
	const gchar *agentKey = "agent";
	node = json_object_get_member(pJsonObject, agentKey);
	if (node !=NULL) {
	

		if (isprimitive("std::string")) {
			jsonToValue(&agent, node, "std::string", "");
		} else {
			
		}
	}
	const gchar *idKey = "id";
	node = json_object_get_member(pJsonObject, idKey);
	if (node !=NULL) {
	

		if (isprimitive("std::string")) {
			jsonToValue(&id, node, "std::string", "");
		} else {
			
		}
	}
	const gchar *nameKey = "name";
	node = json_object_get_member(pJsonObject, nameKey);
	if (node !=NULL) {
	

		if (isprimitive("std::string")) {
			jsonToValue(&name, node, "std::string", "");
		} else {
			
		}
	}
	const gchar *userIdKey = "userId";
	node = json_object_get_member(pJsonObject, userIdKey);
	if (node !=NULL) {
	

		if (isprimitive("std::string")) {
			jsonToValue(&userId, node, "std::string", "");
		} else {
			
		}
	}
	const gchar *createdAtKey = "createdAt";
	node = json_object_get_member(pJsonObject, createdAtKey);
	if (node !=NULL) {
	

		if (isprimitive("long long")) {
			jsonToValue(&createdAt, node, "long long", "");
		} else {
			
		}
	}
	const gchar *legacyProfileKey = "legacyProfile";
	node = json_object_get_member(pJsonObject, legacyProfileKey);
	if (node !=NULL) {
	

		if (isprimitive("bool")) {
			jsonToValue(&legacyProfile, node, "bool", "");
		} else {
			
		}
	}
	const gchar *suspendedKey = "suspended";
	node = json_object_get_member(pJsonObject, suspendedKey);
	if (node !=NULL) {
	

		if (isprimitive("bool")) {
			jsonToValue(&suspended, node, "bool", "");
		} else {
			
		}
	}
	const gchar *paidKey = "paid";
	node = json_object_get_member(pJsonObject, paidKey);
	if (node !=NULL) {
	

		if (isprimitive("bool")) {
			jsonToValue(&paid, node, "bool", "");
		} else {
			
		}
	}
	const gchar *migratedKey = "migrated";
	node = json_object_get_member(pJsonObject, migratedKey);
	if (node !=NULL) {
	

		if (isprimitive("bool")) {
			jsonToValue(&migrated, node, "bool", "");
		} else {
			
		}
	}
	const gchar *legacyKey = "legacy";
	node = json_object_get_member(pJsonObject, legacyKey);
	if (node !=NULL) {
	

		if (isprimitive("bool")) {
			jsonToValue(&legacy, node, "bool", "");
		} else {
			
		}
	}
}

GameProfile::GameProfile(char* json)
{
	this->fromJson(json);
}

char*
GameProfile::toJson()
{
	JsonObject *pJsonObject = json_object_new();
	JsonNode *node;
	if (isprimitive("std::string")) {
		std::string obj = getAgent();
		node = converttoJson(&obj, "std::string", "");
	}
	else {
		
	}
	const gchar *agentKey = "agent";
	json_object_set_member(pJsonObject, agentKey, node);
	if (isprimitive("std::string")) {
		std::string obj = getId();
		node = converttoJson(&obj, "std::string", "");
	}
	else {
		
	}
	const gchar *idKey = "id";
	json_object_set_member(pJsonObject, idKey, node);
	if (isprimitive("std::string")) {
		std::string obj = getName();
		node = converttoJson(&obj, "std::string", "");
	}
	else {
		
	}
	const gchar *nameKey = "name";
	json_object_set_member(pJsonObject, nameKey, node);
	if (isprimitive("std::string")) {
		std::string obj = getUserId();
		node = converttoJson(&obj, "std::string", "");
	}
	else {
		
	}
	const gchar *userIdKey = "userId";
	json_object_set_member(pJsonObject, userIdKey, node);
	if (isprimitive("long long")) {
		long long obj = getCreatedAt();
		node = converttoJson(&obj, "long long", "");
	}
	else {
		
	}
	const gchar *createdAtKey = "createdAt";
	json_object_set_member(pJsonObject, createdAtKey, node);
	if (isprimitive("bool")) {
		bool obj = getLegacyProfile();
		node = converttoJson(&obj, "bool", "");
	}
	else {
		
	}
	const gchar *legacyProfileKey = "legacyProfile";
	json_object_set_member(pJsonObject, legacyProfileKey, node);
	if (isprimitive("bool")) {
		bool obj = getSuspended();
		node = converttoJson(&obj, "bool", "");
	}
	else {
		
	}
	const gchar *suspendedKey = "suspended";
	json_object_set_member(pJsonObject, suspendedKey, node);
	if (isprimitive("bool")) {
		bool obj = getPaid();
		node = converttoJson(&obj, "bool", "");
	}
	else {
		
	}
	const gchar *paidKey = "paid";
	json_object_set_member(pJsonObject, paidKey, node);
	if (isprimitive("bool")) {
		bool obj = getMigrated();
		node = converttoJson(&obj, "bool", "");
	}
	else {
		
	}
	const gchar *migratedKey = "migrated";
	json_object_set_member(pJsonObject, migratedKey, node);
	if (isprimitive("bool")) {
		bool obj = getLegacy();
		node = converttoJson(&obj, "bool", "");
	}
	else {
		
	}
	const gchar *legacyKey = "legacy";
	json_object_set_member(pJsonObject, legacyKey, node);
	node = json_node_alloc();
	json_node_init(node, JSON_NODE_OBJECT);
	json_node_take_object(node, pJsonObject);
	char * ret = json_to_string(node, false);
	json_node_free(node);
	return ret;
}

std::string
GameProfile::getAgent()
{
	return agent;
}

void
GameProfile::setAgent(std::string  agent)
{
	this->agent = agent;
}

std::string
GameProfile::getId()
{
	return id;
}

void
GameProfile::setId(std::string  id)
{
	this->id = id;
}

std::string
GameProfile::getName()
{
	return name;
}

void
GameProfile::setName(std::string  name)
{
	this->name = name;
}

std::string
GameProfile::getUserId()
{
	return userId;
}

void
GameProfile::setUserId(std::string  userId)
{
	this->userId = userId;
}

long long
GameProfile::getCreatedAt()
{
	return createdAt;
}

void
GameProfile::setCreatedAt(long long  createdAt)
{
	this->createdAt = createdAt;
}

bool
GameProfile::getLegacyProfile()
{
	return legacyProfile;
}

void
GameProfile::setLegacyProfile(bool  legacyProfile)
{
	this->legacyProfile = legacyProfile;
}

bool
GameProfile::getSuspended()
{
	return suspended;
}

void
GameProfile::setSuspended(bool  suspended)
{
	this->suspended = suspended;
}

bool
GameProfile::getPaid()
{
	return paid;
}

void
GameProfile::setPaid(bool  paid)
{
	this->paid = paid;
}

bool
GameProfile::getMigrated()
{
	return migrated;
}

void
GameProfile::setMigrated(bool  migrated)
{
	this->migrated = migrated;
}

bool
GameProfile::getLegacy()
{
	return legacy;
}

void
GameProfile::setLegacy(bool  legacy)
{
	this->legacy = legacy;
}


